### PR TITLE
integrate redux with edit page

### PR DESCRIPTION
**Pull request by andrewyanushkevich**

### CODE REVIEW

File: src/common/api/getPasswordsByUserId.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ const getPasswordsByUserId = async userId => {
 
     return response;
   } catch (error) {
-    throw error(error);
+    throw console.error(error);
```
Comment (Dmitriyjd) [inline src/common/api/getPasswordsByUserId.js]: Do we really need to have throw statement here?

File: src/common/pages/edit/components/PasswordForm/component.jsx [hunk 2]
```diff
@@ -25,11 +30,11 @@ class PasswordForm extends PureComponent {
...
           </Form.Item>
-          <Form.Item label="resource">
-            {getFieldDecorator('resource', {
-              initialValue: resource,
+          <Form.Item label="resourceAddress">
+            {getFieldDecorator('resourceAddress', {
+              initialValue: resourceAddress,
               rules: [
-                { required: true, message: 'Please input password resource!' },
+                { required: true, message: 'Please input resource address!' },
```
Comment (Dmitriyjd) [inline src/common/pages/edit/components/PasswordForm/component.jsx]: Move message into constants

File: src/common/pages/edit/containers/index.js [hunk 3]
```diff
@@ -1,17 +1,33 @@
...
 
-const mapStateToProps = state => ({
-  counter: state.edit.count,
-});
+import Edit from 'common/pages/edit/components/PasswordForm';
+import addPassword from 'common/api/addPassword';
+import {
+  addPasswordBegin,
+  addPasswordSuccess,
+  addPasswordFailure,
```
Comment (Dmitriyjd) [inline src/common/pages/edit/containers/index.js:8]: 1. This is editing page, not password creation page.
2. For initial action name use word `Pending`. Like we have pending in `Promises`.

File: src/common/pages/edit/containers/index.js [hunk 4]
```diff
@@ -1,17 +1,33 @@
...
+    dispatch(addPasswordBegin());
+
+    const response = await addPassword(item); // @todo put real user id
+    const { data } = response;
+
+    dispatch(addPasswordSuccess(data));
+  } catch (e) {
+    dispatch(addPasswordFailure(e));
+
+    throw e;
```
Comment (Dmitriyjd) [inline src/common/pages/edit/containers/index.js]: Do we really need to throw something?

File: src/common/pages/edit/containers/index.js [hunk 5]
```diff
@@ -1,17 +1,33 @@
...
 
 const mapDispatchToProps = dispatch => ({
-  increment: () => dispatch(onIncrement()),
-  decrement: () => dispatch(onDecrement()),
+  addPasswordItem: item => dispatch(addPasswordItem(item)),
 });
 
 export default connect(
-  mapStateToProps,
+  undefined,
```
Comment (Dmitriyjd) [inline src/common/pages/edit/containers/index.js]: We need to use `mapStateToProps` function to get original password's params from the store.

File: src/common/store/rootReducer.js [hunk 6]
```diff
@@ -1,10 +1,97 @@
-import { combineReducers } from 'redux';
-import home from 'common/pages/home/store';
-import edit from 'common/pages/edit/store';
+// CONSTS
+const HOME_GET_PASSWORDS_BEGIN = 'HOME_GET_PASSWORDS_BEGIN';
```
Comment (Dmitriyjd) [inline src/common/store/rootReducer.js]: Do no forget to rename begin to pending.

File: src/common/store/rootReducer.js [hunk 7]
```diff
@@ -1,10 +1,97 @@
...
+        error: '',
+        loading: false,
+      };
+    case ADD_PASSWORD_BEGIN:
+      return {
+        ...state,
+        loading: true,
+      };
+    case ADD_PASSWORD_SUCCESS:
+      return Object.assign({}, state, {
```
Comment (Dmitriyjd) [inline src/common/store/rootReducer.js]: Use merging via spread operator instead. { ...state, items: [] }

**Review by Dmitriyjd — CHANGES_REQUESTED** [review]

**Comment by dzianisbohush** [comment]
you will able to find these changes in another pull request

**Pull request closed without merging**